lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -235,14 +270,16 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Value rhs = convertScalarToDtype(b, loc, payloadArgs[1], dtype);
     return b.create<arith::AndIOp>(loc, lhs, rhs);
   }
-  if (isa<AtenLog2Op>(op))
-    return b.create<math::Log2Op>(loc, payloadArgs[0]);
   if (isa<AtenAbsOp>(op))
     return b.create<math::AbsOp>(loc, payloadArgs[0]);
-  if (isa<AtenSigmoidOp>(op)) {
-    Type elementType = payloadArgs[0].getType();
+  if (auto sigmoid = dyn_cast<AtenSigmoidOp>(op)) {
+    Type dtype = converter->convertType(sigmoid.getType())
+                     .cast<RankedTensorType>()
+                     .getElementType(); 
+    Value arg = convertScalarToDtype(b, loc, payloadArgs[0], dtype);
+    Type elementType = arg.getType();
     auto one = b.create<arith::ConstantOp>(loc, FloatAttr::get(elementType, 1));
-    auto negate = b.create<arith::NegFOp>(loc, payloadArgs[0]);
+    auto negate = b.create<arith::NegFOp>(loc, arg);
```

Comment:
nit: You can use the helper function you wrote for the `NegFOp` to avoid doing the `dtype` code above